Kickstarter-iOS/Views/Controllers/CreatePasswordViewController.swift

```diff
@@ -48,13 +69,77 @@ final class CreatePasswordViewController: UITableViewController {
     self.tableView.registerCellClass(SettingsTextInputCell.self)
   }
 
+  override func viewDidAppear(_ animated: Bool) {
+    super.viewDidAppear(animated)
+
+    self.viewModel.inputs.viewDidAppear()
+  }
+
   // MARK: - Styles
 
   override func bindStyles() {
     super.bindStyles()
 
     _ = self.tableView
       |> settingsGroupedTableViewStyle
+      |> \.allowsSelection .~ true
+  }
+
+  // MARK: - Binding
+
+  override func bindViewModel() {
+    super.bindViewModel()
+
+    self.viewModel.outputs.newPasswordConfirmationTextFieldBecomeFirstResponder
+      .observeForUI()
+      .observeValues { [weak self] in
+        self?.newPasswordConfirmationTextField?.becomeFirstResponder()
+    }
+
+    self.viewModel.outputs.newPasswordConfirmationTextFieldResignFirstResponder
+      .observeForUI()
+      .observeValues { [weak self] in
+        self?.newPasswordConfirmationTextField?.resignFirstResponder()
+    }
+
+    self.viewModel.outputs.validationLabelIsHidden
+      .observeForUI()
+      .observeValues { [weak self] isHidden in
+        self?.footerView?.isHidden = isHidden
+    }
+
+    self.viewModel.outputs.validationLabelText
+      .observeForUI()
+      .observeValues { [weak self] text in
+        UIView.performWithoutAnimation {
```

Comment:
It turned out that this was pretty tricky since simply calling `self?.footerView?.configure(with: text)` would not properly resize the footer (if it was initialized without text) - the `label` height would be 0.

To properly reload a footer we have to put the update between `beginUpdates()` & `endUpdates()` which unfortunately had another side effect of animating the footer in and out (this looked quite odd)...turns out we can disable animations for this operation (not globally - so should be safe) by wrapping it inside of `performWithoutAnimation` block.

Oh UIKit sometimes I love you, sometimes not really 😄